### ISSUE TITLE
make install broken

### ISSUE THREAD

**Issue by mikebrow** [kubernetes-incubator/cri-o#326]
If one uses the following commands to build and install, you get an error:
sudo make clean
sudo make
sudo make install

mike@mike-VirtualBox:~/go/src/github.com/kubernetes-incubator/cri-o$ sudo make install
install -D -m 755 /home/mike/go/bin/ocid /usr/local/bin/ocid
install: cannot stat '/home/mike/go/bin/ocid': No such file or directory
Makefile:127: recipe for target 'install' failed
make: *** [install] Error 1

After the following commit:
https://github.com/kubernetes-incubator/cri-o/commit/6c9628cdb11e5720afa6ed07c56683923800125c

the assumption is go build is used to compile not make...

**Comment by runcom** [comment]
@jawnsy would you take a look at this?

**Comment by jawnsy** [comment]
Yup, please assign this to me

**Comment by mikebrow** [comment]
thx jawnsy.. I'm a bit swamped with something else atm

**Comment by jawnsy** [comment]
Hmm. So I tried to reproduce this by running in a clean `centos:7` container: `docker run --rm -it centos:7 /bin/bash` but wasn't successful in doing so.

```
yum install git sudo golang make install btrfs-progs-devel device-mapper-devel glib2-devel glibc-devel gpgme-devel libassuan-devel libgpg-error-devel glibc-static libseccomp-devel
adduser testuser
usermod testuser -aG wheel
passwd testuser # set a password so I can sudo
su - testuser
mkdir -p go/src/github.com/kubernetes-incubator
cd $_
git clone https://github.com/kubernetes-incubator/cri-o.git
cd cri-o
sudo make
sudo make install
```

Seems to run fine, this is the output (note you need `GOPATH=$GOPATH` because `sudo` cleans environment variables for security reasons):

```
[testuser@2465114f1b47 cri-o]$ sudo GOPATH=$GOPATH make
go install \
	-tags "selinux seccomp  " \
	github.com/kubernetes-incubator/cri-o/cmd/ocid
go install github.com/kubernetes-incubator/cri-o/cmd/ocic
go install github.com/kubernetes-incubator/cri-o/cmd/kpod
make -C conmon
make[1]: Entering directory `/home/testuser/go/src/github.com/kubernetes-incubator/cri-o/conmon'
cc -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include     -c -o conmon.o conmon.c
cc -o conmon conmon.o -Wall -Wextra -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include   -lglib-2.0  
make[1]: Leaving directory `/home/testuser/go/src/github.com/kubernetes-incubator/cri-o/conmon'
make -C pause
make[1]: Entering directory `/home/testuser/go/src/github.com/kubernetes-incubator/cri-o/pause'
cc -Os -Wall -Wextra -static   -c -o pause.o pause.c
cc -o pause pause.o -Os -Wall -Wextra -static 
strip pause
make[1]: Leaving directory `/home/testuser/go/src/github.com/kubernetes-incubator/cri-o/pause'
make -C test/bin2img
make[1]: Entering directory `/home/testuser/go/src/github.com/kubernetes-incubator/cri-o/test/bin2img'
go build -o bin2img
make[1]: Leaving directory `/home/testuser/go/src/github.com/kubernetes-incubator/cri-o/test/bin2img'
make -C test/copyimg
make[1]: Entering directory `/home/testuser/go/src/github.com/kubernetes-incubator/cri-o/test/copyimg'
go build -o copyimg
make[1]: Leaving directory `/home/testuser/go/src/github.com/kubernetes-incubator/cri-o/test/copyimg'
make -C test/checkseccomp
make[1]: Entering directory `/home/testuser/go/src/github.com/kubernetes-incubator/cri-o/test/checkseccomp'
make[1]: `checkseccomp' is up to date.
make[1]: Leaving directory `/home/testuser/go/src/github.com/kubernetes-incubator/cri-o/test/checkseccomp'
ocid --config="" config --default > ocid.conf
go-md2man -in docs/ocid.conf.5.md -out docs/ocid.conf.5.tmp && touch docs/ocid.conf.5.tmp && mv docs/ocid.conf.5.tmp docs/ocid.conf.5
go-md2man -in docs/kpod.1.md -out docs/kpod.1.tmp && touch docs/kpod.1.tmp && mv docs/kpod.1.tmp docs/kpod.1
go-md2man -in docs/kpod-launch.1.md -out docs/kpod-launch.1.tmp && touch docs/kpod-launch.1.tmp && mv docs/kpod-launch.1.tmp docs/kpod-launch.1
go-md2man -in docs/ocid.8.md -out docs/ocid.8.tmp && touch docs/ocid.8.tmp && mv docs/ocid.8.tmp docs/ocid.8
[testuser@2465114f1b47 cri-o]$ sudo GOPATH=$GOPATH make install
install -D -m 755 /home/testuser/go/bin/ocid /usr/local/bin/ocid
install -D -m 755 /home/testuser/go/bin/ocic /usr/local/bin/ocic
install -D -m 755 /home/testuser/go/bin/kpod /usr/local/bin/kpod
install -D -m 755 conmon/conmon /usr/local/libexec/ocid/conmon
install -D -m 755 pause/pause /usr/local/libexec/ocid/pause
install -d -m 755 /usr/local/share/man/man1
install -d -m 755 /usr/local/share/man/man5
install -d -m 755 /usr/local/share/man/man8
install -m 644 docs/kpod.1 docs/kpod-launch.1 -t /usr/local/share/man/man1
install -m 644 docs/ocid.conf.5 -t /usr/local/share/man/man5
install -m 644 docs/ocid.8 -t /usr/local/share/man/man8
[testuser@2465114f1b47 cri-o]$ 
```

**Comment by jawnsy** [comment]
I suggest running `make` as your normal user, and only elevating for `make install` or `make localintegration`. It helps minimize the amount of build code that runs as root (who knows if go install does something wacky like tries to install packages to your system GOROOT, and similarly, `make clean` could go haywire and wipe important stuff)

**Comment by mikebrow** [comment]
Hmm.. thx for providing your output.  My sudo also had a valid GOPATH so that wasn't it. 

But after seeing your build output I decided to check my clone directories and git status and boom..there it was. I had some extra directories waiting to be committed. Not sure how I missed that I needed to do a git clean of my clone after merging the last set of commits. I had these files waiting to be added in my cri-o/ clone master:
	kpod/kbod
	ocic/ocic
	ocid/ocid
	vendor/src/

I guess these were no longer needed and that left my git clone in a bad state which somehow resulted in make not running the go install... and thus the make install could not find the source in my go/bin.

So when we do a make it does a go install of the ocid binaries to the developers go/bin? Installing first to go/bin then make install moving it from go/bin to /user/local/bin seems odd and fraught with path issues as man developers place go/bin in their path and this will.. well you get it :) Should probably chat over this path stuff.  But my issue is resolved :)
